storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java

```diff
@@ -187,4 +188,25 @@ public static Long offsetFromFile(File file) {
         return offsetFromFileName(file.getName());
     }
 
+    /**
+     * Invokes every function in `all` even if one or more functions throws an exception.
+     * If any of the functions throws an exception, the first one will be rethrown at the end with subsequent exceptions
+     * added as suppressed exceptions.
+     */
+    public static void tryAll(List<StorageAction<Void, Exception>> all) throws Exception {
```

Comment:
A bit odd to have this method here. This class is meant to have file related utilities and this is not that.